src/formatDate.js

```diff
@@ -9,6 +9,27 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const data = {};
+  const newDateFormat = [];
+  const dateParts = date.split(fromFormat.slice(3));
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    data[fromFormat[i]] = dateParts[i];
+  }
+
+  if (!fromFormat.includes('YY')) {
+    data.YY = data.YYYY.slice(2);
+  } else {
+    const longYearFormat = data.YY < 30 ? 20 + data.YY : 19 + data.YY;
```

Comment:
The `data.YY` is a string, and you are trying to concatenate strings with numbers, which will not result in the correct year format. You need to convert `data.YY` to a number before doing arithmetic operations.